snuba/datasets/entities/discover.py

```diff
@@ -0,0 +1,537 @@
+import logging
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Mapping, Optional, Sequence, Set
+
+from snuba import environment, state
+from snuba.clickhouse.columns import (
+    UUID,
+    Array,
+    ColumnSet,
+    DateTime,
+    FixedString,
+    Float,
+    LowCardinality,
+    Nested,
+    Nullable,
+    String,
+    UInt,
+)
+from snuba.clickhouse.translators.snuba import SnubaClickhouseStrictTranslator
+from snuba.clickhouse.translators.snuba.allowed import (
+    ColumnMapper,
+    SubscriptableReferenceMapper,
+)
+from snuba.clickhouse.translators.snuba.mappers import ColumnToLiteral, ColumnToMapping
+from snuba.clickhouse.translators.snuba.mapping import TranslationMappers
+from snuba.datasets.entity import Entity
+from snuba.datasets.entities import EntityKey
+from snuba.datasets.entities.events import event_translator
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.plans.single_storage import SelectedStorageQueryPlanBuilder
+from snuba.datasets.storage import (
+    QueryStorageSelector,
+    ReadableStorage,
+    StorageAndMappers,
+)
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage
+from snuba.datasets.entities.transactions import transaction_translator
+from snuba.query.conditions import (
+    BINARY_OPERATORS,
+    ConditionFunctions,
+    get_first_level_and_conditions,
+)
+from snuba.query.expressions import Column, Literal, SubscriptableReference
+from snuba.query.extensions import QueryExtension
+from snuba.query.logical import Query
+from snuba.query.matchers import Column as ColumnMatch
+from snuba.query.matchers import FunctionCall as FunctionCallMatch
+from snuba.query.matchers import Literal as LiteralMatch
+from snuba.query.matchers import String as StringMatch
+from snuba.query.matchers import Or, Param
+from snuba.query.parsing import ParsingContext
+from snuba.query.processors import QueryProcessor
+from snuba.query.processors.performance_expressions import apdex_processor
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+from snuba.query.processors.failure_rate_processor import FailureRateProcessor
+from snuba.query.processors.handled_functions import HandledFunctionsProcessor
+from snuba.query.processors.tags_expander import TagsExpanderProcessor
+from snuba.query.processors.timeseries_processor import TimeSeriesProcessor
+from snuba.query.project_extension import ProjectExtension
+from snuba.query.subscripts import subscript_key_column_name
+from snuba.query.timeseries_extension import TimeSeriesExtension
+from snuba.request.request_settings import RequestSettings
+from snuba.util import qualified_column
+from snuba.utils.metrics.backends.wrapper import MetricsWrapper
+
+EVENTS = EntityKey.EVENTS
+TRANSACTIONS = EntityKey.TRANSACTIONS
```

Comment:
These changed type but all your type hints in this file are still based on strings.